src/bin/dual-iir.rs

```diff
@@ -73,8 +75,40 @@ const SAMPLE_TICKS: u32 = 1 << SAMPLE_TICKS_LOG2;
 const SAMPLE_PERIOD: f32 =
     SAMPLE_TICKS as f32 * hardware::design_parameters::TIMER_PERIOD;
 
-#[derive(Clone, Copy, Debug, Tree)]
-pub struct Settings {
+#[derive(Clone, Debug, Tree)]
+pub struct FlashSettings {
+    pub broker: heapless::String<255>,
```

Comment:
I'd put broker, id, mac into a common `struct App { broker: .., id: .., }` and then have the apps impl `trait App { fn app(&self) -> &App; }`. That factors out the common fields and enforces their presence during bootstrap.

Reply:
Modulo bikeshedding on the names...

Reply:
Maybe `Net` for this, `DualIir` for the app settings, and `Settings` for the outermost struct.

Reply:
I'm somewhat confused. This is the outermost struct.

Reply:
`FlashSettings` -> `Settings`
`RuntimeSettings` -> `DualIir`

`NetSettings` already done.

Reply:
Rename finished in c508163325e8f3d7a24149f5be3e54911f68c8b8